cmd/idracsdkapps/elasticsearch/elasticsearchplugin.go

```diff
@@ -0,0 +1,336 @@
+	dataPresent := bytes.Contains(tmp, []byte("data: "))
+	namePresent := bytes.Contains(tmp, []byte("Name"))
+
+	if idPresent == true && dataPresent == true && namePresent == true {
+		return true
+	} else {
+		return false
+	}
+}
+func InsertMetricReportData(es *elasticsearch.Client, index1 int, tmp []byte) {
+	var res Resp
+	mtx.Lock()
+	reportCount++
+	err := json.Unmarshal(tmp, &dat)
+	mtx.Unlock()
+	// NOTE: For the time being I am using the json unmarshall function as a validator for the parsed metric reports
+	if err != nil {
+		// Don't panic, but keep a count of these invalid packets for analysis
+		// panic(err)
+		invalidJsonPkts++
+		return
+	}
+	err = json.Unmarshal(tmp, &res)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("The size of the received buffer in bytes: %d, The number of Metric Reports processed: %d\n", len(tmp), reportCount)
+	esIndex := res.Id
+	fmt.Printf("Metric Report Name: %s\n", esIndex)
+	for _, value := range res.MetricValues {
+		docIndex := value.MetricID
+		//fmt.Printf("MetricId: %s\n", docIndex)
+		fmt.Println("\nDOC _id:", docIndex)
+		//fmt.Println(value)
+		docval, _ := json.Marshal(value)
+		request := esapi.IndexRequest{Index: strings.ToLower(esIndex), Body: strings.NewReader(string(docval)), DocumentID: docIndex, Refresh: "true"}
+		res, err := request.Do(context.Background(), es)
+		if err != nil {
+			fmt.Printf("IndexRequest ERROR: %s\n", err)
+			continue
+		}
+		defer res.Body.Close()
+		if res.IsError() {
+			fmt.Printf("%s ERROR indexing document ID=%s\n", res.Status(), docIndex)
+			fmt.Println("Error str: ", res.String())
+		} else {
+			// Deserialize the response into a map.
+			var resMap map[string]interface{}
+			if err := json.NewDecoder(res.Body).Decode(&resMap); err != nil {
+				fmt.Printf("Error parsing the response body: %s\n", err)
+			} else {
+				fmt.Printf("\nIndexRequest() RESPONSE:\n")
+				// Print the response status and indexed document version.
+				fmt.Println("Status:", res.Status())
+				fmt.Println("Result:", resMap["result"])
+				fmt.Println("Version:", int(resMap["_version"].(float64)))
+				fmt.Println("resMap:", resMap)
+				//fmt.Println("\n")
+			}
+		}
+	}
+	fmt.Println("inserted into ealsticEngine")
+}
+func LoadTelemetryData(es *elasticsearch.Client) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr, Timeout: 10 * time.Second}
+	var outerloop int
+	for {
+		req, err := http.NewRequest("GET", "https://"+telemetryserverIP+"/redfish/v1/SSE?$filter=EventFormatType%20eq%20MetricReport", nil)
+		if err != nil {
+			fmt.Printf("error %s", err)
+			return
+		}
+		req.SetBasicAuth("root", "calvin")
+		req.Header.Add("Accept", "application/json")
+		response, err := client.Do(req)
+		if err != nil {
+			fmt.Printf("error %s", err)
+			return
+		}
+		fmt.Println(response.StatusCode)
+		//body, _ := ioutil.ReadAll(response.Body)
+		defer response.Body.Close()

```

Comment:
bug,  this will only be called when the function exits.  The for loop will lead to multiple connections to be created with no function exit.